aya-obj/src/btf/btf.rs

```diff
@@ -667,13 +675,58 @@ impl BtfExt {
         endianness: Endianness,
         btf: &Btf,
     ) -> Result<BtfExt, BtfError> {
-        // Safety: btf_ext_header is POD so read_unaligned is safe
+        #[repr(C)]
+        #[derive(Debug, Copy, Clone)]
+        struct MinimalHeader {
+            pub magic: u16,
+            pub version: u8,
+            pub flags: u8,
+            pub hdr_len: i32,
+        }
+
+        if data.len() < std::mem::size_of::<MinimalHeader>() {
+            return Err(BtfError::InvalidHeader);
+        }
+
+        // Safety: btf_ext_header and MinimalHeader are POD so read_unaligned is safe
         let header = unsafe {
```

Comment:
please don't use such a large unsafe section, it makes it harder to reason about
what's actually unsafe